enhancements/installer/assisted-installer-bare-metal-validations.md

```diff
@@ -0,0 +1,221 @@
+  installer workflows.
+
+### Non-Goals
+
+- Adding additional validation rules for either the assisted installer
+  or installer-provisioned infrastructure installer.
+- Removing or relaxing any validation rules for the
+  installer-provisioned infrastructure installer.
+- Recapitulate the test plans for the assisted installer.
+
+## Proposal
+
+All of the validation is currently performed during early phases of
+the installer for the `InstallConfig` target. The assisted installer
+does not phase that for the `Cluster` target, so if we move some of
+the validation into that phase we will have a better separation of the
+parts needed to build the inputs used by the assisted installer and
+the parts needed to run the installer for installer-provisioned
+infrastructure for bare metal.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a developer building the assisted installer tools, I want to be
+able to run the OpenShift installer to generate assets for a bare
+metal cluster without invoking validations that only apply to the
+installer-provisioned infrastructure workflows so I can prepare the
+inputs needed for the assisted installation workflow.
+
+#### Story 2
+
+As a developer of the OpenShift installer, I want to be able to add
+validation rules for bare metal clusters correctly based on whether
+the cluster is being built using the installer-provisioned
+infrastructure workflow or the assisted installer workflow so I can
+apply only the rules necessary for each case.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The install config inputs are currently validated as part of
+generating the `InstallConfig` asset in
+[InstallConfig.finish()](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/installconfig.go#L141),
+which invokes some logic that applies to all platforms in
+[ValidateInstallConfig()](https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L40)
+before running the platform-specific logic from
+[validatePlatform()](https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L341). For
+the `baremetal` platform that results in calling
+[ValidatePlatform()](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/validation/platform.go#L218).
+
+The
+[PlatformProvisionCheck](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go)
+and
+[PlatformCredsCheck](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformcredscheck.go)
+assets are both dependencies for the `Cluster` asset, which means they
+are evaluated later in the installation process. Both already have
+platform-specific logic, although not for the `baremetal` platform.
+
+The assisted installer does not need the `bootstrapProvisioningIP`,
+`bootstrapProvisioningIP`, `provisioningNetworkCIDR`,
+`provisioningDHCPRange`, or `provisioningNetworkInterface` fields. The
```

Comment:
What do you think about clusterOSImage and bootstrapOSImage? We could validate that if specified (they're not in the assisted install), that it's a valid URL in the installConfig, and only validate it exists in ProvisionCheck.

Reply:
I added a paragraph about splitting those checks.